torch/_inductor/template_heuristics.py

```diff
@@ -0,0 +1,600 @@
+            {"config": (16, 256, 32, 6, 4), "cond": True},
+            {"config": (16, 32, 64, 6, 2), "cond": True},
+            {"config": (16, 64, 64, 6, 2), "cond": True},
+            {"config": (16, 128, 64, 6, 4), "cond": True},
+            {"config": (16, 256, 64, 6, 4), "cond": True},
+            {"config": (32, 32, 32, 6, 2), "cond": True},
+            {"config": (32, 64, 32, 6, 2), "cond": True},
+            {"config": (32, 128, 32, 6, 4), "cond": True},
+            {"config": (32, 256, 32, 6, 4), "cond": True},
+            {"config": (32, 32, 64, 6, 2), "cond": True},
+            {"config": (32, 64, 64, 6, 2), "cond": True},
+            {"config": (32, 128, 64, 6, 4), "cond": True},
+            {"config": (32, 256, 64, 6, 4), "cond": True},
+        ]
+
+        self.scaled_persistent_mm_configs = [
+            # "BLOCK_M", "BLOCK_N", "BLOCK_K", "num_stages", "num_warps"
+            {"config": (128, 128, 64, 3, 8), "cond": True},
+            {"config": (128, 128, 128, 3, 8), "cond": True},
+            {"config": (128, 128, 128, 4, 8), "cond": True},
+            {"config": (128, 128, 128, 4, 4), "cond": True},
+            {"config": (128, 128, 128, 3, 4), "cond": True},
+            {"config": (128, 128, 128, 5, 4), "cond": True},
+            {"config": (128, 128, 128, 5, 8), "cond": True},
+            {"config": (128, 128, 128, 6, 8), "cond": True},
+            {"config": (128, 128, 64, 4, 8), "cond": True},
+        ]
+
+        # TODO: Unify with other gemm patterns, mm_plus_mm currently follows
+        # slightly different pattern than rest
+        self.mm_plus_mm_configs = [
+            {"config": (64, 64, 32, 2, 4), "cond": True},
+            {"config": (64, 64, 32, 3, 8), "cond": True},
+            {"config": (64, 64, 32, 4, 16), "cond": True},
+            {"config": (64, 32, 32, 4, 8), "cond": True},
+            {"config": (32, 64, 32, 4, 8), "cond": True},
+            {"config": (128, 128, 32, 1, 8), "cond": True},
+            {"config": (64, 64, 64, 1, 8), "cond": True},
+            {"config": (32, 32, 128, 1, 8), "cond": True},
+            {"config": (64, 64, 16, 2, 4), "cond": True},
+            {"config": (32, 32, 16, 1, 2), "cond": True},
+        ]
+
+        self.conv_configs = [
+            # "BLOCK_M", "BLOCK_N", "BLOCK_K", "num_stages", "num_warps"
+            {"config": (64, 256, 16, 2, 4), "cond": True},
+            {"config": (256, 64, 16, 2, 4), "cond": True},
+            {"config": (1024, 16, 16, 1, 8), "cond": True},
+            {"config": (128, 128, 32, 2, 8), "cond": True},
+            {"config": (64, 64, 32, 2, 4), "cond": True},
+            {"config": (64, 256, 32, 2, 8), "cond": True},
+            {"config": (256, 64, 32, 2, 8), "cond": True},
+        ]
+
+    def _filter_configs(self, configs):
+        return tuple(
+            cast(tuple[int, int, int, int, int], config["config"])
+            for config in configs
+            if config["cond"]
```

Comment:
I don't really get this "cond" thing, can you add a docstring explaining it.

Reply:
It's a "simple" way to conditionalise usage of a config. Historically it was used to avoid certain configs on ROCm e.g. 
`{"config": (64, 64, 16, 2, 4), "cond": if not torch.version.hip}`

Keeping this in gives us an ability to define one structure which can filter out certain configs based on hardware specifications, but now we have platform specific heuristic subclasses we could just remove this if you think its cleaner. It's something I was also thinking about removing but left it in for now.

Reply:
This seems redundant with the idea of subclasses.  Is it still used?  I can we delete it?

Reply:
Sure, lets do it!